From 826de6b23f6fd452386bddefde3677b923202b57 Mon Sep 17 00:00:00 2001
From: Lasse Brun <bruners@gmail.com>
Date: Sat, 2 Mar 2013 12:05:30 +0100
Subject: [PATCH] Patch to fix building oss on kernel versions >=3.8

Source: http://www.opensound.com/forum/viewtopic.php?t=5191
Upstream: Patch supplied by upstream on upstream forums
Reason: Fails to build against >3.8 kernels
---
 setup/Linux/oss/build/pci_wrapper.inc | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/setup/Linux/oss/build/pci_wrapper.inc b/setup/Linux/oss/build/pci_wrapper.inc
index e08ec3a..7ddab2e 100644
--- a/setup/Linux/oss/build/pci_wrapper.inc
+++ b/setup/Linux/oss/build/pci_wrapper.inc
@@ -17,7 +17,11 @@ typedef struct
 static dev_map_t dev_map[MAX_INSTANCE];
 static int n_devmap = 0;
 
-static int __devinit
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,8,0)
+  static int __devinit
+#else
+  static int
+#endif
 osspci_probe (struct pci_dev *pcidev, const struct pci_device_id *pciid)
 {
   oss_device_t *osdev;
@@ -65,7 +69,11 @@ osspci_probe (struct pci_dev *pcidev, const struct pci_device_id *pciid)
   return 0;
 }
 
-static void __devexit
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,8,0)
+  static int __devexit
+#else
+  static int
+#endif
 osspci_remove (struct pci_dev *pcidev)
 {
   int i;
@@ -80,10 +88,11 @@ osspci_remove (struct pci_dev *pcidev)
 	pci_disable_device (dev_map[i].pcidev);
 	osdev_delete (osdev);
 
-	return;
+	return 0;
       }
 
   printk (KERN_ALERT DRIVER_NICK ": Can't find the PCI device to detach\n");
+  return -EIO;
 }
 
 void
-- 
1.8.1.3
